server/container/guice/guice-common/src/main/java/org/apache/james/PeriodicalHealthChecksConfiguration.java

```diff
@@ -0,0 +1,108 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james;
+
+import java.time.Duration;
+import java.util.Objects;
+
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.james.util.DurationParser;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+public class PeriodicalHealthChecksConfiguration {
+
+    private static final String HEALTH_CHECK_PERIOD = "healthcheck.period";
+    private static final Duration DEFAULT_HEALTH_CHECK_PERIOD = Duration.ofMinutes(1);
+    private static final Duration MINIMAL_HEALTH_CHECK_PERIOD = Duration.ofSeconds(10);
+    public static final PeriodicalHealthChecksConfiguration DEFAULT_CONFIGURATION = builder()
+        .period(DEFAULT_HEALTH_CHECK_PERIOD)
+        .build();
+
+    public interface Builder {
```

Comment:
Why a builder for a one argument factory  method ?

Reply:
Probably because there was two arguments before and it became one. But I get the point

Reply:
It had more than one argument at first but after changes there's only one left. We might want to add an option to enable/disable logging so i still keep the builder just in case.

Reply:
ok